### ISSUE TITLE
Support access via deps.edn git coordinate

### ISSUE THREAD

**Issue by spencerwilson** [weavejester/cljfmt#288]
Hello, Thanks for your work on cljfmt!

## Problem

I'm eager to get the changes in https://github.com/weavejester/cljfmt/pull/274 but they're not released yet, so I changed tack and attempted a change like the following in my project:
```diff
-           :extra-deps {cljfmt/cljfmt {:mvn/version "0.9.0"}}}
+           :extra-deps {io.github.weavejester/cljfmt {:deps/root "cljfmt" :git/sha "cf708becf6e202ef0bc8c635aebbc49be7cd1c61"}}}
```

That didn't work though, since tools.deps doesn't support as dependencies packages that use Leiningen's "project.clj" as their manifest: https://github.com/clojure/tools.deps/blob/6ae2b6f71773de7549d7f22759e8b09fec27f0d9/src/main/clojure/clojure/tools/deps/extensions.clj#L21

As a result, the Clojure CLI tools don't know how to resolve the above dependency:
```
% clj -M:cljfmt --help
Error building classpath. Manifest file not found for io.github.weavejester/cljfmt in coordinate {:deps/root "cljfmt", :git/sha "cf708becf6e202ef0bc8c635aebbc49be7cd1c61", :git/url "https://github.com/weavejester/cljfmt.git"}
```

## Suggested solution

Maybe have both a project.clj and deps.edn? The former would remain authoritative for development on cljfmt, whereas the latter would be for the benefit of teaching tools.deps what it needs to know to load the source tree as a dependency. This approach risks drift between the two manifests, and would increase maintenance burden.

**Comment by spencerwilson** [comment]
Ah, @seancorfield suggested a workaround here: https://ask.clojure.org/index.php/8507/failure-on-github-import-via-deps

**Comment by spencerwilson** [comment]
Workaround didn't actually suffice: `readresource` macroexpansion here e.g., https://github.com/weavejester/cljfmt/blob/9de4fab1dfac5c6c05d574dede670a823d499265/cljfmt/src/cljfmt/core.cljc#L292

would fail due to the [indents](https://github.com/weavejester/cljfmt/tree/9de4fab1dfac5c6c05d574dede670a823d499265/cljfmt/resources/cljfmt/indents) resources not being on the classpath. I'm not sure if it's tools.deps who, when handed a git dependency, isn't putting the resources on the classpath that it sets up, or what. Am somewhat out of my depth here and had to move on.

**Comment by weavejester** [comment]
Fixed by #293.

**Comment by borkdude** [comment]
Indeed:

``` clojure
 bb -Sdeps '{:deps {cljfmt/cljfmt {:git/url "https://github.com/weavejester/cljfmt" :git/sha "7dfd55d5dd0756f30311a90f206c2dd32e56d18b" :deps/root "cljfmt"}}}' -m cljfmt.main
```

works fine now.
